src/renderer/services/authService/mockAuthClient.ts

```diff
@@ -0,0 +1,60 @@
+import { delay } from "@common/delay";
+import multicast from "observable-fns/multicast";
+import Subject from "observable-fns/subject";
+
+import type { AuthService, AuthUser } from "./types";
+
+export class MockAuthClient implements AuthService {
+  private _currentUser: AuthUser | null = null;
+  private _userSubject = new Subject<AuthUser | null>();
+  private _onAuthStateChanged = multicast(this._userSubject);
+
+  public async init(): Promise<AuthUser | null> {
+    await delay(3000);
```

Comment:
is it a necessity to delay on init?

Reply:
Not necessary, but it's good for testing/simulating loading states etc.